### PR TITLE
[USM] reduce process exit events cpu pressure in callbacks

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

As we are running in callback spawned by process monitor
reduce the lock pressure as multiple callback can call the same code path
and likely(!found) as pid exit event are not related to ssl or gotls

### Motivation

Avoid mutex/cpu pressure

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/protocols/http/ebpf_gotls.go [hunk 1]
```diff
@@ -396,13 +396,16 @@ func (p *GoTLSProgram) registerProcess(binID binaryID, pid pid, mTime syscall.Ti
 }
 
 func (p *GoTLSProgram) unregisterProcess(pid pid) {
-	p.lock.Lock()
-	defer p.lock.Unlock()
-
+	p.lock.RLock()
```
Comment (guyarb) [inline pkg/network/protocols/http/ebpf_gotls.go:399]: what about javatls?

**Review by guyarb — APPROVED** [review]

File: pkg/network/protocols/http/ebpf_gotls.go (hunk 1 above)
Comment (nplanel) [reply pkg/network/protocols/http/ebpf_gotls.go:399]: we only subscribe on exec event on javatls

**Review by guyarb — APPROVED** [review]

File: pkg/network/usm/ebpf_gotls.go [hunk 2]
```diff
@@ -396,13 +396,15 @@ func (p *GoTLSProgram) registerProcess(binID binaryID, pid pid, mTime syscall.Ti
...
-
+	p.lock.RLock()
 	binID, found := p.processes[pid]
+	p.lock.RUnlock()
 	if !found {
 		return
 	}
+
+	p.lock.Lock()
+	defer p.lock.Unlock()
```
Comment (guyarb) [inline pkg/network/usm/ebpf_gotls.go:407]: in a second review, the pattern can be problematic
and we need to add:
```
p.lock.Lock()
defer p.lock.Unlock()
binID, found := p.processes[pid]
if !found {
	return
}
```

File: pkg/network/usm/shared_libraries.go [hunk 3]
```diff
@@ -264,13 +264,15 @@ func (r *soRegistry) cleanup() {
...
-
+	r.m.RLock()
 	paths, found := r.byPID[pid]
+	r.m.RUnlock()
 	if !found {
 		return
 	}
+
+	r.m.Lock()
+	defer r.m.Unlock()
```
Comment (guyarb) [inline pkg/network/usm/shared_libraries.go:275]: in a second review, the pattern can be problematic
and we need to add:
```
p.lock.Lock()
defer p.lock.Unlock()
binID, found := p.processes[pid]
if !found {
	return
}
```

**Review by guyarb — CHANGES_REQUESTED** [review]

**Review by guyarb — APPROVED** [review]

File: pkg/network/usm/ebpf_gotls.go (hunk 2 above)
Comment (nplanel) [reply pkg/network/usm/ebpf_gotls.go]: good catch

**Pull request merged**